src/stdlib/stdlib/std/internal/math.tact

```diff
@@ -430,3 +430,34 @@ fun sqrt(num: Int): Int {
 
     return x;
 }
+
+/// Global function. Available since Tact 1.6.0.
+///
+/// Computes the integer square root of the `Int` value `num`. Returns the rounded down result.
+/// This is equivalent to finding the largest integer `x` such that `x * x ≤ num`.
+///
+/// Attempts to specify a negative value of `num` throw an exception with exit code 5: `Integer out of expected range`.
+///
+/// ```tact
+/// fun examples() {
+///     isqrt(4);  // 2
+///     isqrt(3);  // 1
+///     isqrt(2);  // 1
+///     isqrt(1);  // 1
+///     isqrt(0);  // 0
+///     isqrt(-1); // ERROR! Exit code 5: Integer out of expected range
+/// }
+/// ```
+///
+/// See: https://docs.tact-lang.org/ref/core-math#isqrt
+///
+fun isqrt(num: Int): Int {
+    let x = sqrt(num);
+
+    if (x > num / x) {
```

Comment:
it's not clear why `isqrt(0) == 0` and not an exception

Reply:
why should it be an exception?

Reply:
`let x = sqrt(0)`, so x = 0 and `num / x` should produce an exception